DatadogInternal/Sources/NetworkInstrumentation/URLSession/URLSessionTaskDelegateSwizzler.swift

```diff
@@ -8,105 +8,95 @@ import Foundation
 
 /// Swizzles `URLSessionTaskDelegate` callbacks.
 internal class URLSessionTaskDelegateSwizzler {
-    private static var _didFinishCollectingMap: [String: DidFinishCollecting?] = [:]
-    static var didFinishCollectingMap: [String: DidFinishCollecting?] {
-        get {
-            lock.lock()
-            defer { lock.unlock() }
-            return _didFinishCollectingMap
-        }
-        set {
-            lock.lock()
-            defer { lock.unlock() }
-            _didFinishCollectingMap = newValue
-        }
-    }
-    private static var lock = NSRecursiveLock()
-
-    private static var _didCompleteWithErrorMap: [String: DidCompleteWithError?] = [:]
-    static var didCompleteWithErrorMap: [String: DidCompleteWithError?] {
-        get {
-            lock.lock()
-            defer { lock.unlock() }
-            return _didCompleteWithErrorMap
-        }
-        set {
-            lock.lock()
-            defer { lock.unlock() }
-            _didCompleteWithErrorMap = newValue
-        }
-    }
+    private let lock: NSLocking
+    private var didReceive: DidReceive?
```

Comment:
didReceive is only available in `URLSessionDataDelegate`

https://developer.apple.com/documentation/foundation/urlsessiondatadelegate/1410027-urlsession

It should be in its own swizzler instance.

Reply:
ah yea 👍